src/wallet/interfaces.cpp

```diff
@@ -318,13 +318,12 @@ class WalletImpl : public Wallet
         }
         return {};
     }
-    std::vector<WalletTx> getWalletTxs() override
+    std::set<WalletTx> getWalletTxs() override
     {
         LOCK(m_wallet->cs_wallet);
-        std::vector<WalletTx> result;
-        result.reserve(m_wallet->mapWallet.size());
+        std::set<WalletTx> result;
         for (const auto& entry : m_wallet->mapWallet) {
-            result.emplace_back(MakeWalletTx(*m_wallet, entry.second));
+            result.emplace(MakeWalletTx(*m_wallet, entry.second));
```

Comment:
d3a6bba3fdeba44a630f14755f73a9cb1a301b09

This seems to be a bad change for the GUI when loading a big wallet.

The order by hash requirement comes from `TransactionTablePriv::updateWallet` implementation, definitely something to improve/refactor.

Reply:
I think it won't be too bad, but fixing that seems like a bigger change than suitable for this PR.